executor/src/witgen/query_processor.rs

```diff
@@ -117,11 +117,19 @@ where
             }
             Expression::Reference(Reference::Poly(poly)) => {
                 if !poly.next && poly.index.is_none() {
-                    let poly_id = poly.poly_id();
+                    let poly_id = poly.poly_id.unwrap();
                     match poly_id.ptype {
-                        PolynomialType::Committed | PolynomialType::Intermediate => rows
-                            .get_value(poly)
-                            .ok_or(IncompleteCause::DataNotYetAvailable),
+                        PolynomialType::Committed | PolynomialType::Intermediate => {
+                            let poly_ref = AlgebraicReference {
```

Comment:
At this point, `PolynomialReference` leaks into witgen code, because it is part of a `parsed::Expression` reference. I think it should be converted to an `AlgebraicReference` before, no?

Reply:
The problem is that the query needs to be a full expression and thus needs a proper PolynomialReference and not the restricted algebraic one.